### DISCUSSION TITLE
Автоматическая подстановка id треков в Source.getRecomTracks

### DISCUSSION THREAD

**Discussion by Bedmit** [Chimildic/goofy#23] [Помощь с алгоритмом] [answered]
Можно ли каким то образом не вписывать идентификатор (например трека), а чтобы он сам брался из какого-либо источника?
Например в скрипте 
```
  let Test = Source.getRecomTracks({
      seed_tracks: 'absd',
});
```
чтобы подставлялся допустим первый трек определенного плейлиста.

**Comment by Chimildic** [comment, marked as answer]
Через _Selector_ ограничить количество. Через _map_ создать массив с `id`. Через _join_ создать строчку с разделением через запятую.

```js
let tracks = Source.getPlaylistTracks('name', 'id');
Selector.keepRandom(tracks, 5);
let recomTracks = Source.getRecomTracks({
    seed_tracks: tracks.map(t => t.id).join(','),
});
```

**Reply by Bedmit** [reply]
Спасибо. Ничего не понял, но просто скопировал, заменив _Selector_ на _First_. Получилось то что и хотел.
Хорошо бы это в инструкцию добавить, наверное.

**Comment by Bedmit** [comment]
Сделал радио по треку. Поставил триггер на час.
Добавляем в плейлист **Radio** любой трек и этот плейлист становится постоянно генерируемым радио с фильтрами по этому треку, до тех пор пока не добавить в него другой трек.
```
function Radio() {

let OneTrack = Source.getPlaylistTracks('Radio', '7KwPNhl8wnnbDqGL4i7QEK');
Selector.keepLast(OneTrack, 1);
let Radio = Source.getRecomTracks({
    seed_tracks: OneTrack.map(t => t.id).join(','),
});

  Filter.dedupArtists(Radio);
  let Past = Source.getSavedTracks();
  Filter.removeTracks(Radio, Past);
  let History = RecentTracks.get();
  Filter.removeTracks(Radio, History);
  Filter.rangeDateRel(History, 1);
  Filter.removeArtists(Radio, History); 
  Combiner.push(Radio, OneTrack);

  let UpdTime = 'Обновлено в ' + new Date().toTimeString().substring(0, 5);

  Playlist.saveWithReplace({
  id: '7KwPNhl8wnnbDqGL4i7QEK',
  name: 'Radio',
  description: UpdTime,
  tracks: Radio,
});
}
```

**Reply by Chimildic** [reply]
Если это проблема, то я не понял ее описание

**Reply by Bedmit** [reply]
Да это я просто результатом поделился
